packages/backend/src/apps/postman/actions/send-transactional-email/index.ts

```diff
@@ -30,17 +31,16 @@ export default defineAction({
     if (!result.success) {
       throw fromZodError((result as SafeParseError<unknown>).error)
     }
+    let recipients = result.data.destinationEmail.slice(+progress)
+    recipients = await getRatelimitedRecipientList(recipients)
+    const newProgress = +progress + recipients.length
 
-    const results = await sendTransactionalEmails(
-      $.http,
-      result.data.destinationEmail,
-      {
-        subject: result.data.subject,
-        body: result.data.body,
-        replyTo: result.data.replyTo,
-        senderName: result.data.senderName,
-      },
-    )
+    const results = await sendTransactionalEmails($.http, recipients, {
```

Comment:
There's a very, very small edge case where we will still get rate limit by postman:
1. Redis has problem and `await emailRateCounter.get('send')` returns `NaN`, so we will default to 169 rate limit.
2. Two workers receive action to email 1000000 people; they call `getRatelimitedRecipientList` and get 169 each.
3. The workers calls to`sendTransactionalEmails` start failing after postman rate limits us.


I know in #233 you did auto-retry for connectivity issues only; do we also want to:
1. Handle this via auto-retry
2. Handle this explicitly (e.g. try catch, then increase progress only for succeeded emails)
3. Don't handle at all

Reply:
@ogp-weeloong This is won't happened as `emailRateCounter.consume` will fail for one of them (as per the rate limiter) and be caught to return an empty array (i.e. additional progress `0`). This case has been tested under the test case `2 concurrent email pipelines with workload both higher than rate limit`